uvicgse/project-virtual-team-2
# Feature: Improve stash: drop/branch

Follow-up from #102 

**DROP**
This branch has functionality that implements `git stash drop` from #35 and #84 

Test its functionality by dropping stashes from the stash list in the stash modal and check if it was successful from the command line or in the VisGit tree

This branch also contains the updated stash modal, a hotfix for the recent repository display, and a hotfix for properly updating the event/error modal for git actions

**To review**: Create a stash entry and drop it. Confirm with the command line `git stash list` and working tree (stashed node should be gone and stash modal list should be updated accordingly)

**BRANCH**

This feature allows the user to create a new branch from the commit of a stash and then applies the stash on top of the new branch. This is done by opening the stash-modal from the nav-bar, clicking on the dropdown menu of a particular stash, and clicking on the branch button. This will open the branch modal where you can name, create, and checkout the branch. The user can also open the branch-modal and create a new branch from the most recent stash at index 0.

This PR also recolors the stash nodes in the graph to the same orange as the modified files in the file-panel to indicate that it is a stash and not a commit.

To test this feature:

have or create a stash
create a branch from the stash as described above
confirm its success by examining the branch menu, branch tags in the graph, and the terminal and log for accuracy.
confirm that the stash files were re-applied properly
